libusb/os/darwin_usb.c

```diff
@@ -2263,15 +2263,17 @@ static int darwin_abort_transfers (struct usbi_transfer *itransfer) {
   /* abort transactions */
 #if InterfaceVersion >= 550
   if (LIBUSB_TRANSFER_TYPE_BULK_STREAM == transfer->type)
-    (*(cInterface->interface))->AbortStreamsPipe (cInterface->interface, pipeRef, itransfer->stream_id);
+    kresult = (*(cInterface->interface))->AbortStreamsPipe (cInterface->interface, pipeRef, itransfer->stream_id);
   else
 #endif
-    (*(cInterface->interface))->AbortPipe (cInterface->interface, pipeRef);
+    kresult = (*(cInterface->interface))->AbortPipe (cInterface->interface, pipeRef);
 
-  usbi_dbg (ctx, "calling clear pipe stall to clear the data toggle bit");
-
-  /* newer versions of darwin support clearing additional bits on the device's endpoint */
+#if InterfaceVersion <= 245
```

Comment:
Is this is right test?  Shouldn't it be `MAC_OS_X_VERSION_MIN_REQUIRED < 1050`?

If you build with the 10.6 SDK, InterfaceVersion will be >= 245, but you may still have a deployment target of 10.5.

Reply:
I don't think either is actually the right test, because what we really want is the version in use at runtime.

Reply:
Given that we're talking about very old macOS versions, I think just altering the compile-time check is better. A runtime check would incur a tiny performance hit, and just about everyone is using >= 10.5 these days.

Reply:
Do we really care about 10.5 at all?
macports is supporting old version but only down to 10.6 Snow Leopard.
https://ports.macports.org/port/libusb/details/

Reply:
I think it's nice to keep 10.5 support, it's the last version to support PowerPC. The codebase already supports it. All this PR needs is a one line change:

`#if InterfaceVersion <= 245`

to

`MAC_OS_X_VERSION_MIN_REQUIRED < 1050`

Reply:
I see. In that case, no problem from my side.

Just wondering if anyone has really tested libusb on macOS 10.5 on Power PC though.

Reply:
I think Nathan's reasoning is that building for older releases with newer SDK is difficult, or impossible, to get right, and the old SDKs should be used when building for older releases.

Anyway I think we can merge this as-is for a start, and then fix up the version checks systematically in all places.